### PR TITLE
parser: support of character in create table statement

### LINKED ISSUE #7371: Syntax support of character in create table statement
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
create table t (c1 NATIONAL CHARACTER(10));
```

2. What did you expect to see?
Table created.

3. What did you see instead?
ERROR 1105 (HY000): line 1 column 39 near "(10))" (total length 44)

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
87625923e0943923b00b3242a143afa02a455385

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Fix #7371 to make some JDBC test cases happy.

### What is changed and how it works?
Support `NATIONAL CHARACTER` type in create table statement, like `create table t (c1 NATIONAL CHARACTER(10));` by add parser rules. Because `NATIONAL CHARACTER` is same as `NATIONAL CHAR`, just change parser is okay.

Reference link: https://dev.mysql.com/doc/refman/5.7/en/charset-national.html

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects
no

Related changes
no

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -6094,6 +6094,17 @@ StringType:
 		}
 		$$ = x
 	}
+|	"NATIONAL" "CHARACTER" FieldLen OptBinary OptCollate
```
Comment (shenli) [inline parser/parser.y]: Can we define a new parser rule like:
```
CharOrCharacter:
     "CHAR" | "CHARACTER"
```
Then we could merge this into the first branch.

Comment (imtbkcat) [reply parser/parser.y]: @shenli 
NationalOpt could be empty,  "CHAR(10)" is legal, but "CHARACTER(10)" is illegal in MySQL. There must be a "NATIONAL" before "CHARACTER".

**Comment by coocood** [comment]
LGTM

File: parser/parser.y (hunk 1 above)
Comment (shenli) [reply parser/parser.y]: OK

**Pull request merged**